wallet-next/src/worker.rs

```diff
@@ -1,21 +1,29 @@
+use std::sync::{Arc, Mutex};
+
 use crate::{sync::scan_block, Storage};
 use penumbra_crypto::{merkle::NoteCommitmentTree, FullViewingKey};
 use penumbra_proto::client::oblivious::{
     oblivious_query_client::ObliviousQueryClient, CompactBlockRangeRequest,
 };
+use tokio::sync::mpsc::{error::TryRecvError, Receiver};
 use tonic::transport::Channel;
 #[derive(Clone)]
 pub struct Worker {
     storage: Storage,
     client: ObliviousQueryClient<Channel>,
     nct: NoteCommitmentTree,
     fvk: FullViewingKey, // TODO: notifications (see TODOs on WalletService)
+    error_slot: Arc<Mutex<Option<anyhow::Error>>>,
+    // This is in an `Arc` because `Receiver` is not `Clone`.
```

Comment:
Why does Receiver need to be Clone? We only have one worker, right?

Reply:
The Worker type had an existing Clone annotation.

Reply:
Ah, cool. I don't think we need it -- we consume the worker to run it, so it'd be better to drop the Clone bound and hold the receiver directly.